fix(sn_networking): actually retry PUTs

joshuef opened the pull request:
## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 06 Oct 23 18:16 UTC
This pull request fixes the issue where PUT requests were not being retried. The patch adds a check to see if the response matches certain error conditions, and if so, continues the retry loop.
<!-- reviewpad:summarize:end -->

joshuef committed:
fix(sn_networking): actually retry PUTs

RolandSherwin commented on sn_networking/src/lib.rs:
```diff
@@ -385,6 +385,13 @@ impl Network {
+                || matches!(res, Err(Error::RecordNotFound))
+            {
+                continue;
+            }
```
I don't think we propagate the `RecordNotEnoughCopies` uptill here? It is intercepted and modified during `get_record`.

Also should we perform `retries+1` here to prevent infinte loop?

RolandSherwin commented on sn_networking/src/lib.rs:
```diff
@@ -385,6 +385,13 @@ impl Network {
             let res = self
                 .put_record_once(record.clone(), verify_store.clone())
                 .await;
+
```
Also this is performed only if `verify_store.is_some()`, so maybe add a comment stating that or maybe add in `if verify_storel.is_some() {}` to indicate it?

joshuef committed:
chore(sn_networking): ensure we return on put_record

if we're not verifying
or its okay, we exit,
otherwise increase retry count and try again
